src/app/_components/CheckBox/CheckBox.js

```diff
@@ -0,0 +1,172 @@
+'use client';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import cn from '@/utils/cn';
+import { checkBoxVariants } from './styles';
+
+export default function CheckBox({
+  value,
+  onChange,
+  name,
+  text = '',
+  subText = '',
+  disabled = false,
+  checked = false,
+  children = null,
+  variant = checkBoxVariants.default,
+}) {
+  const id = `checkbox_id_${name}_|_${value}__`;
+  return (
+    <div className={cn(variant.container.base)}>
+      <input
+        id={id}
+        name={name}
+        className={cn(
+          variant.checkBox.base,
+          variant.checkBox.hover,
+          variant.checkBox.focus,
+          variant.checkBox.focusRing,
+          variant.checkBox.checked,
+          variant.checkBox.checkedFocus,
+          variant.checkBox.checkedHover,
+          variant.checkBox.disabled,
+        )}
+        value={value}
+        onChange={onChange}
+        type="checkbox"
+        checked={checked}
+        disabled={disabled}
+      />
+      <label
+        className={cn(
+          variant.labelBefore.base,
+          variant.labelBefore.layout,
+          variant.labelBefore.peerFocus,
+          variant.labelBefore.peerChecked,
+          variant.labelBefore.peerDisabled,
+        )}
+        htmlFor={id}
+      >
+        {checked && (
+          <Tick
+            styles={cn(
+              variant.tick.position,
+              variant.tick.base,
+              disabled && variant.tick.disabled,
+            )}
+          />
+        )}
+        {text && (
+          <div
+            className={cn(
+              variant.textContainer.base,
+              variant.textContainer.position,
+            )}
+          >
+            {text ? (
+              <>
+                {text && (
+                  <p
+                    className={cn(
+                      variant.text.base,
+                      disabled && variant.text.disabled,
+                    )}
+                  >
+                    {text}
+                  </p>
+                )}
+                {subText && (
+                  <p
+                    className={cn(
+                      variant.subText.base,
+                      disabled && variant.subText.disabled,
+                    )}
+                  >
+                    {subText}
+                  </p>
+                )}
+              </>
+            ) : (
+              { children }
+            )}
+          </div>
+        )}
+      </label>
+    </div>
+  );
+}
+
+function Tick({ styles }) {
```

Comment:
@IvanKoshan Consider moving the Tick component to a separate file (`Tick.js`) within the same directory to make code clear, possibility to reuse Tick component.

Reply:
moved tick icon to dedicated icons folder

Reply:
Thanks! Then use it in this file instead of this function.

Reply:
omg, did i really just leave it in there like that? Fixed it for sure now